addons/gddialog/ui/ui_database_editor/ui_hbc_filters/ui-hbc-filters.gd

```diff
@@ -0,0 +1,107 @@
+@tool
+extends HBoxContainer
+
+@onready var nSignals: Node = get_node( "Signals" )
+@onready var nLineEditSearch: LineEdit = get_node( "LineEditSearch" )
+@onready var nOptionButtonSearchBy: OptionButton = get_node( 
+		"OptionButtonSearchBy" )
+@onready var nCheckBoxTagFilter: CheckBox = get_node( "CheckBoxTagFilter" )
+@onready var nMenuButtonFilterTags: MenuButton = get_node( 
+		"MenuButtonFilterTags" )
+
+
+""" Search Features """
+
+
+func set_all_records_visible( value: bool ) -> void:
+	var records = owner.nVBCDialogRecords.get_children()
+	for record in records:
+		record.visible = value
+
+
+func update_filter() -> void:
+	var filter_text: String = nLineEditSearch.text
+	set_all_records_visible( true )
+	#	Fill an array with the tags we have selected to filter by.
+	var selected_tags: Array
+	var nFilterMenu: PopupMenu = nMenuButtonFilterTags.get_popup()
+	for tag in range( nFilterMenu.get_item_count() ):
+		if( nFilterMenu.is_item_checked( tag ) ):
+			var tag_name: String = nFilterMenu.get_item_text( tag )
+			selected_tags.push_back( tag_name )
+	#	Now we gather the records we will be filtering.
+	var records: Array = owner.nVBCDialogRecords.get_children()
+	#	Are we filtering by Name/Description/ID/etc?
+	var filter_option: int = nOptionButtonSearchBy.selected
+	#	Now to do the record filtering.
+	for record in records:
+		var id: int = record.get_record_id()
+		var matched: bool = true
+		if( filter_text.length() != 0 ):
+			matched = false
+			match filter_option:
+				0: #	Name
+					if( owner.nDatabase.database[ id ].has( "name" ) ):
+						#	Search the name for the string.
+						var record_name = (
+								owner.nDatabase.database[ id ][ "name" ] )
+						if( record_name.find( filter_text ) > -1 ):
+							matched = true
+				1: #	Description
+					var record_desc = (
+							owner.nDatabase.database[ id ][ "description" ] )
+					if( record_desc.find( filter_text ) > -1 ):
+							matched = true
+				2: #	ID
+					if( str( id ) == filter_text ):
+						matched = true
+		#	Of the ones that match the above result, we want to filter out
+		#	 records that don't fit any of the selected tags.
+		if( matched == true
+			and nCheckBoxTagFilter.button_pressed == true
+			and nFilterMenu.get_item_count() > 1
+		):
+			matched = false
+			#	"Include Untagged" means show the ones with no tags at all.
```

Comment:
Unneeded. It doesn't need to be explained and if it did, it shouldn't be done here.

Reply:
Reduce all unneeded in this function.